cmd/kubeapps-apis/plugins/kapp_controller/packages/v1alpha1/server_ctrl_packages.go

```diff
@@ -612,8 +612,18 @@ func (s *Server) UpdateInstalledPackage(ctx context.Context, request *corev1.Upd
 		return nil, statuserror.FromK8sError("get", "PackageInstall", installedPackageName, err)
 	}
 
+	versionConstraints, err := versionConstraintWithPolicy(pkgVersion, s.upgradePolicy)
+	if err != nil {
+		return nil, err
+	}
+
 	// Update the rest of the fields
-	pkgInstall.Spec.PackageRef.VersionSelection = &vendirVersions.VersionSelectionSemver{Constraints: pkgVersion}
+	pkgInstall.Spec.PackageRef.VersionSelection = &vendirversions.VersionSelectionSemver{
+		Constraints: versionConstraints,
+		// https://github.com/vmware-tanzu/carvel-kapp-controller/issues/116
+		// This is to allow prereleases to be also installed
```

Comment:
I'm not certain from the comment exactly what the decision is here? The issue has been closed (fixed almost a year ago), but from the description seems to only be relevant *if* you want to include prereleases in your version selection? Also, see the later https://github.com/vmware-tanzu/carvel-kapp-controller/issues/300 . So wouldn't that mean that we're making that decision here to include pre-releases, as opposed to not doing so by default?

Looks to me like another config option for `defaultIncludePrereleases` or similar?

Reply:
Right... I missed this pending task when defining the list of pending actions. I initially add this `VersionSelectionSemverPrereleases{}` to explicitly allow prereleases by default, which is the opposite behavior in kapp-ctrl, just to ensure we supported everything (it's easier to notice something is there when it shouldn't than the other way round :P )

Since it's a simple and similar change as the one I did here, let me merge this PR and create a follow-up one straight away.
Thanks for pointing this out!

Reply:
Done: https://github.com/kubeapps/kubeapps/pull/4165